#! /bin/sh /usr/share/dpatch/dpatch-run
## 91_trim-fix.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Solution from CVS for bug #178
## DP: ( http://vdr-developer.org/mantisbt/view.php?id=178 )

@DPATCH@
diff -urNad vdr-plugin-burn-0.1.0~pre21~/common.c vdr-plugin-burn-0.1.0~pre21/common.c
--- vdr-plugin-burn-0.1.0~pre21~/common.c	2006-09-16 20:33:36.000000000 +0200
+++ vdr-plugin-burn-0.1.0~pre21/common.c	2006-10-31 16:57:55.000000000 +0100
@@ -2,7 +2,7 @@
  * See the files COPYING and README for copyright information and how to reach
  * the author.
  *
- *  $Id: common.c,v 1.28 2006/09/16 18:33:36 lordjaxom Exp $
+ *  $Id: common.c,v 1.29 2006/10/01 21:22:27 lordjaxom Exp $
  */
 
 #include "burn.h"
@@ -116,15 +116,11 @@
 	return builder.str();
 }
 
-	void trim_left( std::string& text_, const char* characters_ )
+	void trim_left( std::string& text_, const char* characters_, std::string::size_type offset_ )
 	{
-		std::string::size_type pos = 0;
-		do {
-			while ( strchr( characters_, text_[ pos ] ) != NULL )
-				text_.erase( 0, 1 );
-			if ( ( pos = text_.find( '~' ) ) != std::string::npos )
-				++pos;
-		} while ( pos != std::string::npos );
+	    std::string::size_type pos;
+	    if ( ( pos = text_.find_first_not_of( characters_, offset_ ) ) > 0 )
+	        text_.erase( offset_, pos - offset_ );
 	}
 
 string get_recording_datetime(const cRecording* recording_, char delimiter)
diff -urNad vdr-plugin-burn-0.1.0~pre21~/common.h vdr-plugin-burn-0.1.0~pre21/common.h
--- vdr-plugin-burn-0.1.0~pre21~/common.h	2006-09-16 20:33:36.000000000 +0200
+++ vdr-plugin-burn-0.1.0~pre21/common.h	2006-10-31 16:57:55.000000000 +0100
@@ -2,7 +2,7 @@
  * See the files COPYING and README for copyright information and how to reach
  * the author.
  *
- *  $Id: common.h,v 1.28 2006/09/16 18:33:36 lordjaxom Exp $
+ *  $Id: common.h,v 1.29 2006/10/01 21:22:27 lordjaxom Exp $
  */
 
 #ifndef VDR_BURN_COMMON_H
@@ -194,7 +194,7 @@
 
 int ScanPageCount(const std::string& Path);
 std::string progress_bar(double current, double total, int length = 20);
-	void trim_left( std::string& text_, const char* characters_ );
+	void trim_left( std::string& text_, const char* characters_, std::string::size_type offset_ = 0 );
 
 	//!--- recording helper functions -----------------------------------------
 
diff -urNad vdr-plugin-burn-0.1.0~pre21~/jobs.c vdr-plugin-burn-0.1.0~pre21/jobs.c
--- vdr-plugin-burn-0.1.0~pre21~/jobs.c	2006-09-16 20:33:36.000000000 +0200
+++ vdr-plugin-burn-0.1.0~pre21/jobs.c	2006-10-31 16:57:55.000000000 +0100
@@ -2,7 +2,7 @@
  * See the files COPYING and README for copyright information and how to reach
  * the author.
  *
- *  $Id: jobs.c,v 1.34 2006/09/16 18:33:36 lordjaxom Exp $
+ *  $Id: jobs.c,v 1.35 2006/10/01 21:22:27 lordjaxom Exp $
  */
 
 #include "burn.h"
@@ -47,15 +47,19 @@
 			m_fileName( recording_->FileName() ),
 			m_summary( get_recording_description(recording_) ),
 			m_datetime( get_recording_datetime(recording_, ' ') ),
+			m_title( recording_->Name() ),
 			m_totalSize( 0, 0 ),
 			m_totalLength( 0, 0 )
 	{
-		string title = recording_->Name();
-		string::size_type pos;
-		if (global_setup().RemovePath && (pos = title.rfind('~')) != string::npos)
-			title.erase(0, pos + 1);
-		trim_left(title, "%@");
-		m_title = title;
+		std::string::size_type pos;
+		if ( global_setup().RemovePath && ( pos = m_title.rfind( '~' ) ) != std::string::npos )
+			m_title.erase( 0, pos + 1 );
+
+        for ( pos = 0;; ++pos ) {
+            trim_left( m_title, "%@", pos );
+            if ( ( pos = m_title.find( '~', pos ) ) == std::string::npos )
+                break;
+        }
 	}
 
 //	size_pair::size_type recording::get_total_size() const
